vtex-apps/store-image
# Promotion events

#### What problem is this solving?

Make it possible to send Analytics events when seeing and clicking in a image.

Promotion View: https://developers.google.com/tag-manager/enhanced-ecommerce#promo-impressions
Promotion Click: https://developers.google.com/tag-manager/enhanced-ecommerce#promo-clicks

#### How to test it?

You can check this workspace:
https://promoview--storecomponents.myvtex.com/

I added the promotion props to the images in the images of the slider, the The Office background on top and the The Office background on the bottom.

The event is only triggered when the image is in the viewport.

You can check for the events typing: `pixelManagerEvents` in the Chrome Dev Tools Console.
![image](https://user-images.githubusercontent.com/284515/98573709-e9a04b80-2295-11eb-9454-9ab09d9c01cb.png)

#### Screenshots or example usage:

I added some optional fields to the image schema that one can add via Site Editor:
https://promoview--storecomponents.myvtex.com/admin/cms/site-editor
![image](https://user-images.githubusercontent.com/284515/98573306-78f92f00-2295-11eb-9222-d74ec2c2e596.png)

#### Describe alternatives you've considered, if any.

It was based on this PR #28 by @mariusfiliutamindcell 

#### Related to / Depends on

We should also create a PR in the Google Tag Manager app to trigger this event to GTM.

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/xUySTTDhOw4LF9Esvu/giphy.gif)
